tests/connect/conftest.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+import pytest
+from pyspark.sql import SparkSession
+
+
+@pytest.fixture(scope="session")
+def spark_session():
+    """
+    Fixture to create and clean up a Spark session.
+
+    This fixture is available to all test files and creates a single
+    Spark session for the entire test suite run.
+    """
+    from daft.daft import connect_start
+
+    # Start Daft Connect server
+    server = connect_start("sc://localhost:50051")
```

Comment:
instead of a hardcoded port, we should use a randomly unassigned port. This should prevent the need to add a `time.sleep` to reclaim the port. 

by using `0` for the port it should auto assign it.

maybe something like 

```py
(server, port) = connect_start() # port is auto assigned
(server, _port) = connect_start("sc://localhost:50051") # manually assigned

session = ...
```

Reply:
well would `(server, port) = connect_start()` bind to `0.0.0.0:0`? 

- 👍 really easy
- 👎potential security issue although not really sure if it matters 

anyway hoping cfd62bde0df0de4debf7d10d09d19845cb478006 resolves

Reply:
i think randomly assigning a port is fine. If the user wants a specific port, then they can provide it easily enough. 

This appears to be how lakesail does it too https://docs.lakesail.com/sail/latest/guide/getting-started/#using-the-sail-library

Reply:
I just wanna clarify I mean the IP is the vulnerability (i.e., `0.0.0.0` say if we in reality just want localhost)

what are your thoughts?